dashboard/app/models/user_script.rb

```diff
@@ -36,4 +38,17 @@ def empty?
     # a user script is empty if there is no progress
     started_at.nil? && completed_at.nil? && assigned_at.nil? && last_progress_at.nil?
   end
+
+  def check_plc_task_assignment_updating
+    if check_completed? && self.script.pd
+      #Get all the task assignments for the user for script level tasks
+
+      #I'm not wild about the joinery here - we can definitely save time by adding denormalizing user_id on the
+      #enrollment_task_assignment table. The number of script completions tasks associated with a user won't be that
+      #high, so it's okay to select all of them and pick the ones we want
+      task_assignments_for_user = (Plc::UserCourseEnrollment.where(user: user).map {|x| x.plc_task_assignments.all}).flatten
+      tasks_to_update = task_assignments_for_user.select {|x| x.plc_task.type == 'Plc::ScriptCompletionTask' && x.plc_task.script_id == script.id}
+      tasks_to_update.map(&:complete_assignment)
```

Comment:
`task_assignments_for_user` is asking to be defined as a has_many and has_many :through on user. Then if you did something like:
`user.plc_task_assignments.where(type: 'Plc::ScriptCompletionTask', script_id: script_id).update(status :completed)` it should be 'optimized' by activerecord to look like @wjordan's query. If you think you are using `user.plc_task_assignments.where(type: 'Plc::ScriptCompletionTask)` in multiple places you could make that a helper method on user also.

(this would also make the interface on User the same if you add the denormalized user_id column later, it will just be a change from has_many :through to has_many)